File: 3685-count-subarrays-of-length-three-with-a-condition/3685-count-subarrays-of-length-three-with-a-condition.cpp
class Solution {
   public:
    int countSubarrays(vector<int>& nums) {
        int ans = 0;
        int n = nums.size();

        for (int i = 1; i < n - 1; i++) {
            if (2 * (nums[i - 1] + nums[i + 1]) == nums[i]) ans++;
        }

        return ans;
    }
};

File: 3685-count-subarrays-of-length-three-with-a-condition/3685-count-subarrays-of-length-three-with-a-condition.go
func countSubarrays(nums []int) int {
    ans := 0
    for i := 1; i < len(nums)-1; i++{ 
        if 2*(nums[i-1] + nums[i+1]) == nums[i] {
            ans++
        }
    }
    return ans
}